### DISCUSSION TITLE
How to call Camera::world_to_screen()?

### DISCUSSION THREAD

**Discussion by davidrusu** [bevyengine/bevy#4640] [Q&A] [answered]
In bevy 0.6.0, I was converting from world space to screen space using something like:

```rust
fn my_system(windows: Res<Windows>, camera_query: Query<(&Camera, &GlobalTransform)>) {
    if let Some((camera, cam_trans)) = camera_query.iter().next() {
        // ...
        let screen_point = camera.world_to_screen(&windows, &cam_trans, world_point);
        // ...
    }
}
```

But in 0.7.0, `Camera::world_to_screen()` has changed to also take an `Assets<Image>` parameter: https://docs.rs/bevy/latest/bevy/render/camera/struct.Camera.html#method.world_to_screen

I can't figure out where I might get this `Assets<Image>` parameter from. `Assets` has no obvious constructors, and I don't see any functions that return `Assets<..>`.

**Comment by alice-i-cecile** [comment, marked as answer]
You can fetch `Assets` from the `World` as a resource. The standard asset collections (like `Image`) are initialized by the `DefaultPlugins`.

**Reply by cart** [reply]
As a heads up, this api will be much simpler to call when we land #4745

```rust
camera.world_to_view(&camera_transform, world_position);
```

**Comment by davidrusu** [comment]
Thanks! finally got back to this now, just to expand on your answer slightly:

I added `images: Res<Assets<bevy::prelude::Image>>` to my system arguments and passed `&images` to `camera.world_to_screen(&windows, &images, &cam_trans, world_point)` and that got things running for me again.
